runtime/integration-tests/src/generic/cases/lp/mod.rs

```diff
@@ -0,0 +1,1953 @@
+		CurrencyId::ForeignAsset(100_002)
+	}
+
+	fn ed(&self) -> Balance {
+		100_000_000_000_000
+	}
+
+	fn decimals(&self) -> u32 {
+		18
+	}
+}
+
+#[allow(non_camel_case_types)]
+pub struct FRAX;
+impl CurrencyInfo for FRAX {
+	fn custom(&self) -> CustomMetadata {
+		CustomMetadata {
+			pool_currency: true,
+			transferability: CrossChainTransferability::LiquidityPools,
+			permissioned: false,
+			mintable: false,
+			local_representation: None,
+		}
+	}
+
+	fn symbol(&self) -> &'static str {
+		"FRAX"
+	}
+
+	fn id(&self) -> CurrencyId {
+		CurrencyId::ForeignAsset(100_003)
+	}
+
+	fn ed(&self) -> Balance {
+		100_000_000_000_000
+	}
+
+	fn decimals(&self) -> u32 {
+		18
+	}
+}
+
+#[allow(non_camel_case_types)]
+pub struct LocalUSDC;
+impl CurrencyInfo for LocalUSDC {
+	fn custom(&self) -> CustomMetadata {
+		CustomMetadata {
+			pool_currency: true,
+			transferability: CrossChainTransferability::None,
+			permissioned: false,
+			mintable: false,
+			local_representation: None,
+		}
+	}
+
+	fn symbol(&self) -> &'static str {
+		"LocalUSDC"
+	}
+
+	fn id(&self) -> CurrencyId {
+		CurrencyId::LocalAsset(LOCAL_ASSET_ID)
+	}
+
+	fn ed(&self) -> Balance {
+		10_000
+	}
+
+	fn decimals(&self) -> u32 {
+		6
+	}
+}
+
+/// The faked router address on the EVM side. Needed for the precompile to
+/// verify the origin of messages.
+///
+/// NOTE: This is NOT the real address of the
+///       router, but the one we are faking on the EVM side. Hence, it is fix
+///       coded here in the same way it is fixed code on the EVM testing router.
+pub const EVM_LP_INSTANCE: [u8; 20] = hex!("1111111111111111111111111111111111111111");
+
+/// The faked domain name the LP messages are coming from and going to.
+pub const EVM_DOMAIN_STR: &str = "TestDomain";
+
+/// The test domain ChainId for the tests.
+pub const EVM_DOMAIN_CHAIN_ID: u64 = 1;
+
+pub const EVM_DOMAIN: Domain = Domain::EVM(EVM_DOMAIN_CHAIN_ID);
+
+pub fn setup_full<T: Runtime>() -> impl EnvEvmExtension<T> {
+	setup::<T, _>(|evm| {
+		setup_currencies(evm);
+		setup_pools(evm);
+		setup_tranches(evm);
+		setup_investment_currencies(evm);
+		setup_deploy_lps(evm);
+		setup_investors(evm)
+	})
+}
+
+/// Default setup required for EVM <> CFG communication
+pub fn setup<T: Runtime, F: FnOnce(&mut <RuntimeEnv<T> as EnvEvmExtension<T>>::EvmEnv)>(
+	additional: F,
+) -> impl EnvEvmExtension<T> {
+	let mut env = RuntimeEnv::<T>::from_parachain_storage(
+		Genesis::default()
+			.add(genesis::balances::<T>(DEFAULT_BALANCE * CFG))
+			.storage(),
+	);
+	env.state_mut(|evm| {
+		evm_balances::<T>(DEFAULT_BALANCE * CFG);
+		set_order_book_feeder::<T>(T::RuntimeOriginExt::root());
+
+		evm.load_contracts();
+
+		// Fund gateway sender
+		give_balance::<T>(
+			<T as pallet_liquidity_pools_gateway::Config>::Sender::get(),
+			DEFAULT_BALANCE * CFG,
+		);
+
+		// Register general local pool-currency
+		register_currency::<T>(LocalUSDC, |_| {});
+
+		/* TODO: Use that but index needed contracts afterwards
```

Comment:
Nit - do we want this now or should we have a follow-up for this?

Reply:
Follow up. It requires setting up quite a few hacky contracts for foundry. Possible, but hard. Would of course make it possible for us to not have to refactor the huge list of deployments below all the time, but something for the next iteration.